src/http/api/account.ts

```diff
@@ -234,119 +236,91 @@ export function createGetAccountPostsHandler(
         }
 
         const logger = ctx.get('logger');
-        let account: Account | null = null;
-        const db = ctx.get('db');
-
-        const apCtx = fedify.createContext(ctx.req.raw as Request, {
-            db,
-            globaldb: ctx.get('globaldb'),
-            logger,
-        });
+        const site = ctx.get('site');
 
         const handle = ctx.req.param('handle');
         if (!handle) {
             return new Response(null, { status: 400 });
         }
 
-        const defaultAccount = await accountRepository.getBySite(
-            ctx.get('site'),
-        );
+        const defaultAccount = await accountRepository.getBySite(site);
+
+        let accountPosts: AccountPosts;
 
         if (!defaultAccount || !defaultAccount.id) {
-            return new Response(null, { status: 400 });
+            return new Response(null, { status: 404 });
         }
 
         // We are using the keyword 'me', if we want to get the posts of the current user
         if (handle === 'me') {
-            account = defaultAccount;
+            accountPosts = await accountPostsView.getPostsByAccount(
+                defaultAccount.id,
+                defaultAccount.id,
+                params.limit,
+                params.cursor,
+            );
         } else {
-            if (!isHandle(handle)) {
-                return new Response(null, { status: 400 });
-            }
+            const ctx = fedifyContextFactory.getFedifyContext();
+            const apId = await lookupAPIdByHandle(ctx, handle);
 
-            const apId = await lookupAPIdByHandle(apCtx, handle);
-            if (apId) {
-                account = await accountRepository.getByApId(new URL(apId));
+            if (!apId) {
+                return new Response(`AP ID not found for handle: ${handle}`, {
+                    status: 400,
+                });
             }
-        }
 
-        const result: GetProfileDataResult = {
-            results: [],
-            nextCursor: null,
-        };
-
-        try {
-            //If we found the account in our db and it's an internal account, do an internal lookup
-            if (account?.isInternal && account.id) {
-                const postResult = await postService.getPostsByAccount(
-                    account.id,
-                    defaultAccount.id,
-                    params.limit,
-                    params.cursor,
-                );
+            const account = await accountRepository.getByApId(new URL(apId));
+            if (!account) {
+                return new Response(`Account not found for handle: ${handle}`, {
```

Comment:
Does this mean that we error for all remote accounts that we don't have a record of in our db? Am I misunderstanding this?

Reply:
hmm..this doesn't sound right. I'll check. Thanks for pointing it out.

Reply:
Done. Removed this check